lib/ui/text.dart

```diff
@@ -644,6 +656,58 @@ class ParagraphStyle {
   }
 }
 
+/// A single text shadow.
+/// 
+/// Multiple shadows are stacked together in a [TextStyle].
+class TextShadow {
```

Comment:
why TextShadow rather than Shadow? (Using Shadow would let BoxShadow inherit from it too.)

Reply:
This class is the dart:ui equivalent to shadows, and is only used for text. I didn't want the framework Shadow class be dependent on the dart:ui class that really only exists as a bridge from flutter to dart ui.

See shadow.dart and box_shadow.dart on the framework side for the inherited/generic version

Reply:
I don't really see why we wouldn't want BoxShadow to inherit from the dart:ui class? Can you elaborate on that?

Reply:
I was thinking that the shadow class in dart:ui would provide things like serialization that would not be useful to the framework as a whole. Such methods would probably just add clutter to the API, as it is very specific to the use case of passing data to LibTxt. Also, I figured the framework itself could use the existence of a simple shadow class alongside the more advanced BoxShadow. 

From a user's perspective, I think it makes more sense for them to import the painting package and have a Shadow than importing dart:ui for Shadow.

Reply:
The privates would be private, so the subclasses would be unaffected by them.
The painting package would re-export Shadow, they wouldn't know that it was there.

The problem with having three classes is that it's one third as many classes. It's also confusing that you create a "Shadow" to style text, but at some point it becomes a "TextShadow". Also that "Shadow" is the superclass of "BoxShadow" but not "TextShadow".

As far as I can tell, TextShadow is a strict subset of BoxShadow, so it seems to make sense to me to say that BoxShadow is-a TextShadow (renamed accordingly to Shadow).

Reply:
Ok, that makes sense. I'll make the changes!